### PR TITLE
enable retries upon job failure to increase resources automatically. Also bump nextflow version and conda declarations to nf-core latest

### PR DESCRIPTION
**Pull request by taylorreiter**
FastQC keeps failing for some of the long reads, either with not enough time or OOM. I think enabling retries and scaling resources with the attempt number should fix this.

Edit: adding details requested by @elizabethmcd on nf-core/conda:
In december, nf-core switched how conda packages are declared at the top of modules. This coincided with a nextflow version bump. if you don’t bump your nextflow but do change your conda declarations, docker profiles fail to run. At first this was only failing on CI for me, not locally, because conda updates the nextflow version automatically if you bump the version on nf-core tools. but to switch the nextflow version in CI, you not only need to change it in the github workflow but also in the README and the nextflow config

### CODE REVIEW

**Review by elizabethmcd — APPROVED** [review]

**Comment by taylorreiter** [comment]
well -- CI is going to fail bc multiqc doesn't have a conda package bc the sourmash modules haven't been merged yet. sighhhh

Edit: I decided to enable conda from an earlier version because we're releasing this as a pub soon, and I want tests to be passing/the pipeline to be functional/all the juicy stuff on the master branch when we do that.

**Comment by elizabethmcd** [comment]
Do I need to re-review this or is there a different PR with these updates?

**Comment by taylorreiter** [comment]
re-review please! the first time you reviewed it was failing all of its tests

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,6 +11,10 @@ on:
 env:
   NXF_ANSI_LOG: false
 
+concurrency:
```
Comment (elizabethmcd) [inline .github/workflows/ci.yml:14]: What does this do?

File: README.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ On release, automated continuous integration tests run the pipeline on a full-si
 
 ## Quick Start
 
-1. Install [`Nextflow`](https://www.nextflow.io/docs/latest/getstarted.html#installation) (`>=21.10.3`)
+1. Install [`Nextflow`](https://www.nextflow.io/docs/latest/getstarted.html#installation) (`>=22.10.1`)
```
Comment (elizabethmcd) [inline README.md:21]: we should make these small changes to the nextflow github template we have

File: modules.json [hunk 3]
```diff
@@ -7,22 +7,12 @@
                 "nf-core": {
                     "custom/dumpsoftwareversions": {
                         "branch": "master",
-                        "git_sha": "5e34754d42cd2d5d248ca8673c0a53cdf5624905",
+                        "git_sha": "c8e35eb2055c099720a75538d1b8adb3fb5a464c",
```
Comment (elizabethmcd) [inline modules.json:10]: are these updated nf-core modules?

File: lib/WorkflowMain.groovy [hunk 4]
```diff
@@ -66,7 +66,7 @@ class WorkflowMain {
         NfcoreTemplate.checkConfigProvided(workflow, log)
 
         // Check that conda channels are set-up correctly
-        if (params.enable_conda) {
+        if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
```
Comment (elizabethmcd) [inline lib/WorkflowMain.groovy:69]: what does this do? I know you posted quite a few Nextflow/CI learnings in slack to get these working, particularly for conda

File: nextflow.config [hunk 5]
```diff
@@ -74,15 +72,15 @@ try {
 profiles {
     debug { process.beforeScript = 'echo $HOSTNAME' }
     conda {
-        params.enable_conda    = true
+        conda.enabled          = true
```
Comment (elizabethmcd) [inline nextflow.config:75]: what does this do by changing from a parameter to true by default? is it so the CI tests will run?

File: nextflow_schema.json [hunk 6]
```diff
@@ -226,12 +226,6 @@
                     "description": "Show all params when using `--help`",
                     "hidden": true,
                     "help_text": "By default, parameters set as _hidden_ in the schema are not shown on the command line when a user runs with `--help`. Specifying this option will tell the pipeline to show all parameters."
-                },
-                "enable_conda": {
```
Comment (elizabethmcd) [inline nextflow_schema.json:230]: is this the change nf-core did by no longer using this functionality?

**Review by elizabethmcd — APPROVED** [review]
Could you add a general comment that matches your software channel Slack on the many learnings that seem to be related to the fixes in this PR? Additionally it seems there are quite a lot of changes in the dumpsoftwareversions template python script...did Mert do this for the phylorthology pipeline or was it a change made by nf-core? 
And then not directly related to this PR, I think it would be very helpful to open issues in the nextflow template cookiecutter repo what changes need to be made so we don't have the same downstream CI issues

File: .github/workflows/ci.yml (hunk 1 above)
Comment (taylorreiter) [reply .github/workflows/ci.yml:14]: this is an nf-core change that I updated to

File: modules.json (hunk 3 above)
Comment (taylorreiter) [reply modules.json:10]: yes

File: lib/WorkflowMain.groovy (hunk 4 above)
Comment (taylorreiter) [reply lib/WorkflowMain.groovy:69]: ya this is another of the nf-core changes that enables the conda declarations

File: nextflow.config (hunk 5 above)
Comment (taylorreiter) [reply nextflow.config:75]: this is another nf-core change for conda declarations

File: nextflow_schema.json (hunk 6 above)
Comment (taylorreiter) [reply nextflow_schema.json:230]: this is one of them yes

**Pull request merged**